### ISSUE TITLE
Store Opening Hours are not displayed for German Language

### ISSUE THREAD

**Issue by janwidmer** [SAP/spartacus#11367]
**Describe the bug**
With ticket #10302, the storefinder opening hours where fixed to work for languages other than english. The fix works for e.g. french and italian, but not for german. 

As it seems like, since Java 9, the Date Short Name contains a dot after the weekday abbreviation, e.g. "Mo." for Monday. (See https://stackoverflow.com/a/46245412/2477219) 

The problem is, that the code used in the frontend to match the weekday coming from the backend, does not add such a dot and therefore is not able to match the weekday and therefore shows "-" for the opening hours:

```
// method getSchedule in store-data.service.ts
// weekDayOpeningListItem.weekDay comes with "Mo." from the backend
// date.toLocaleString(language as string, { weekday: 'short', }) returns "Mo" in the frontend
// therefore comparison fails and not even the english hours are being displayed
weekDayOpeningListItem.weekDay === date.toLocaleString(language as string, { weekday: 'short', })
```

**Tell us the version of Spartacus**
- Library version: 3.1.0

**To Reproduce**
1. Use a fresh instance of Spartacus
2. Start it in SSR Mode
3. Open the Store Finder Detail page in German in an ingcognito window (where you have never selected a language via Language Dropdown)
4. See the opening hours not being displayed, but just "-"

**Expected behavior**
The opening hours work also for the German language

**Desktop (please complete the following information):**
 - OS: all
 - Browser all
 - Version all

**Smartphone (please complete the following information):**
 - Device: all
 - OS: all
 - Browser all
 - Version all

**Additional context**
I do not fully understand, why the mapping of the already translated weekday from the backend is done in the frontend.. why are not just the weekdays coming from the backend displayed with there opening hours also already coming in the correct format?

As workaround, one could use the `COMPAT` mode in the backend to still get the weekday without a ".", otherwise one could remove the "." in the frontend before calling the above code in the `store-data.service.ts`, but both workarounds are rather hacky..

Please also see ticket #11366, which is kinda related..

**Comment by Parthlakhani** [comment]
Closing the issue as it is fixed by #11442 
Refer to similar issue #11366

**Comment by giancorderoortiz** [comment]
Retest done with develop branch (May 17, 2021), after 3.1.0. 

![Screen Shot 2021-05-18 at 8 04 00 AM](https://user-images.githubusercontent.com/46171897/118648798-cc05ee80-b7b0-11eb-81db-70f1f2d587ff.png)
